InvenTree/part/models.py

```diff
@@ -68,28 +67,47 @@ def delete(self, *args, **kwargs):
 
         This must be handled within a transaction.atomic(), otherwise the tree structure is damaged
         """
-        with transaction.atomic():
+        print(kwargs)
+        try:
+            if kwargs.get('no_rebuild') is None:
```

Comment:
You make this comparison multiple times, I would suggest doing:

```rebuild = kwargs.get('rebuild', True)```

And then use the `rebuild` argument, and pass `rebuild=False` to disable rebuild